sourced/ml/core/algorithms/token_parser.py

```diff
@@ -76,6 +80,23 @@ def max_token_length(self, value):
             raise ValueError("max_token_length must be greater than 0 - got %d" % value)
         self._max_token_length = value
 
+    @property
+    def use_nn(self):
+        return self._use_nn
+
+    @use_nn.setter
+    def use_nn(self, value):
+        self._init_nn(value)
+
+    def _init_nn(self, value):
+        if value:
+            self._use_nn = True
+            from sourced.ml.core.models.id_splitter import IdentifierSplitterBiLSTM
+            import modelforge.backends
+            backend = modelforge.backends.create_backend()
+            self._id_splitter_nn = IdentifierSplitterBiLSTM().load(
+                "522bdd11-d1fa-49dd-9e51-87c529283418")
```

Comment:
This UUID should be configurable in `__init__`. E.g.

`__init__(nn_model: Optional[str] = None)`

and then remove `use_nn` from the args.

Reply:
But what if we want to use default NN model ?

Reply:
I mean doesn't it forces the user to specify an UUID if he wants to use nn parser ?

Reply:
Sure thing, but you don't have to specify a uuid at all - it will take the default one from the index.